galasa-ui/src/tests/headerMenu.test.tsx

```diff
@@ -0,0 +1,121 @@
+import { render, screen, fireEvent, waitFor, act, getByTestId } from '@testing-library/react';
+import { useRouter } from 'next/navigation';
+import PageHeaderMenu from '@/components/PageHeaderMenu';
+import PageHeader from '@/components/PageHeader';
+import React, { useState } from 'react';
+import { ToastNotification } from '@carbon/react';
+
+const fetchMock = jest.spyOn(global, 'fetch')
+
+let setState : any;
+jest.mock('react', () => ({
+    ...jest.requireActual('react'),
+    useState: jest.fn()
+}))
```

Comment:
Don't think we need to be mocking out useState here - we would want to check the result displayed if useState had an impact anywhere